### PR TITLE
Adding Glo-90

### LINKED ISSUE #34: Missing glo-30 DEM tiles over Azerbaijan, Armenia
### The bug
Glo-30 tiles have missing tiles over said areas. There are downstream implications related to removing topographic phase for GUNWs.

A reference to some of the missing tiles: https://twitter.com/EricFielding/status/1531639815610830848

```
Country identifier | GeoCell ID
Armenia | N38E046
Armenia | N39E044
Armenia | N39E045
Armenia | N39E046
Armenia | N40E043
Armenia | N40E044
Armenia | N40E045
Armenia | N41E043
Armenia | N41E044
Armenia | N41E045
Azerbaijan | N38E046
Azerbaijan | N39E045
Azerbaijan | N39E046
Azerbaijan | N39E047
Azerbaijan | N40E045
Azerbaijan | N40E046
Azerbaijan | N40E047
Azerbaijan | N38E045
Azerbaijan | N38E046
Azerbaijan | N38E048
Azerbaijan | N38E049
Azerbaijan | N39E044
Azerbaijan | N39E045
Azerbaijan | N39E046
Azerbaijan | N39E047
Azerbaijan | N39E048
Azerbaijan | N39E049
Azerbaijan | N40E044
Azerbaijan | N40E045
Azerbaijan | N40E046
Azerbaijan | N40E047
Azerbaijan | N40E048
Azerbaijan | N40E049
Azerbaijan | N40E050
Azerbaijan | N41E044
Azerbaijan | N41E045
Azerbaijan | N41E046
Azerbaijan | N41E047
Azerbaijan | N41E048
Azerbaijan | N41E049
```

### To Reproduce
Check this package data [file](https://github.com/ACCESS-Cloud-Based-InSAR/dem-stitcher/blob/dev/dem_stitcher/data/glo_30.geojson.zip) and verify above tiles doe not exist

 ### Additional context
Downstream topographic phase removal - see Issue [#67](https://github.com/ACCESS-Cloud-Based-InSAR/DockerizedTopsApp/issues/67#issue-1254481042)

### PR DESCRIPTION
**Pull request by cmarshak**
Helps resolve #34 by adding `glo_90` and geojson with missing tiles. Demonstration on how to merge data together (in memory).

### CODE REVIEW

**Comment by cmarshak** [comment]
The main changes of this request are:

1. Allow `glo_30` to be filled in by `glo_90` in the API
2. Add exceptions that catch: a) no available tiles of specifed DEM, b) badly specified `dem_name` and c) extent/bounds not of the form `xmin, ymin, xmax, ymax`.
3. Relevant tests for the above

The notebooks that illustrate the filling is here: https://github.com/ACCESS-Cloud-Based-InSAR/dem-stitcher/blob/glo_90/notebooks/Filling_in_missing_glo-30_tiles_with_glo-90_tiles.ipynb

The final output can be found at this [link](https://gunw-development-testing.s3.us-west-2.amazonaws.com/glo-30-filling/glo_30_filled_in.tif).

A few more things:

1. The `glo_90` AWS page does not distribute a shapefile of all possible tiles anymore so we now go through the bucket to find available tiles. The relevant notebook has been updated. Previously, there were tiles across the globe all listed and many were not available.
2. We have a dataset `glo_90_missing` which can be downloaded [here](https://github.com/ACCESS-Cloud-Based-InSAR/dem-stitcher/blob/glo_90/dem_stitcher/data/glo_90_missing.geojson). These are `glo_30` tiles that are not available but whose corresponding tile as `glo_90` is.

**Review by sssangha — APPROVED** [review]
Looks good to me

File: dem_stitcher/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class NoDEMCoverage(Exception):
+    "Throw this exception if extent/bounds does not cover any available DEM tiles"
+    pass
```
Comment (jhkennedy) [inline dem_stitcher/exceptions.py:3]:
```suggestion
    """Throw this exception if extent/bounds does not cover any available DEM tiles"""
```
As long as you have _docstrings_, you don't need the `pass`. For example:
https://github.com/ASFHyP3/hyp3-sdk/blob/develop/hyp3_sdk/exceptions.py#L7-L8

File: dem_stitcher/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+
+class DEMNotSupported(Exception):
+    "Throw this exception if DEM Name is not supported"
+    pass
```
Comment (jhkennedy) [inline dem_stitcher/exceptions.py:8]:
```suggestion
    """Throw this exception if DEM Name is not supported"""
```

**Review by jhkennedy — APPROVED** [review]
This looks good to me! I haven't run this myself, assuming @cmarshak and @sssangha have confirmed the DEM looks correct.

Couple super minor suggestions; approval stand w/ or w/o making them.

**Pull request merged**